src/ast/passes/codegen_llvm.cpp

```diff
@@ -1301,7 +1301,23 @@ void CodegenLLVM::visit(Call &call)
     b_.CreateOutput(ctx_, buf, getStructSize(event_struct), &call.loc);
     b_.CreateLifetimeEnd(buf);
     expr_ = nullptr;
-  } else if (call.func == "len") {
+  } else if (call.func == "len" && call.vargs.at(0)->type.IsStack()) {
+    auto *arg = call.vargs.at(0);
+    auto scoped_del = accept(arg);
+
+    std::vector<llvm::Type *> stack_key_elements = {
```

Comment:
can we extract this out into a helper e.g. `getStackType` so it's not duplicated here?